### PR TITLE
Add cert-rotation configmap for setting base rotation interval

### PR DESCRIPTION
**Pull request by soltysh**
/assign @deads2k

### CODE REVIEW

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -139,12 +140,31 @@ func RunOperator(ctx *controllercmd.ControllerContext) error {
 		ctx.EventRecorder,
 	)
 
+	certRotationBase := time.Duration(0)
+	err = wait.PollImmediate(time.Second, 1*time.Minute, func() (bool, error) {
```
Comment (deads2k) [inline pkg/operator/starter.go]: has this leaked other places?  smells like a library function if we're going to do it this way.

File: pkg/operator/certrotationcontroller/certrotationcontroller.go [hunk 2]
```diff
@@ -22,17 +27,25 @@ func NewCertRotationController(
...
+
 	certRotator, err := certrotation.NewCertRotationController(
 		"CSRSigningCert",
 		certrotation.SigningRotation{
 			Namespace: operatorclient.OperatorNamespace,
 			// this is not a typo, this is the signer of the signer
 			Name:          "csr-signer-signer",
-			Validity:      8 * time.Hour, // to be 10 days
-			Refresh:       4 * time.Hour, // to be 4 days
+			Validity:      30 * rotationDay,
```
Comment (deads2k) [inline pkg/operator/certrotationcontroller/certrotationcontroller.go]: I think this one needs to be 60 days and 30 days

File: pkg/operator/certrotationcontroller/certrotationcontroller.go [hunk 3]
```diff
@@ -49,8 +62,8 @@ func NewCertRotationController(
 		certrotation.TargetRotation{
 			Namespace: operatorclient.OperatorNamespace,
 			Name:      "csr-signer",
-			Validity:  1 * 4 * time.Hour, // to be 5 days
-			Refresh:   2 * time.Hour,     // to be 1 day
+			Validity:  30 * rotationDay,
+			Refresh:   15 * rotationDay,
```
Comment (deads2k) [inline pkg/operator/certrotationcontroller/certrotationcontroller.go]: any reason not to rotate this every day?

Comment (deads2k) [reply pkg/operator/certrotationcontroller/certrotationcontroller.go]: > any reason not to rotate this every day?

oh, you're chasing the restart.  Someone look to see if we can just make it reload serving trust dynamically?

File: pkg/operator/starter.go [hunk 4]
```diff
@@ -139,12 +140,31 @@ func RunOperator(ctx *controllercmd.ControllerContext) error {
...
+		certRotationConfig, err := kubeClient.CoreV1().ConfigMaps(operatorclient.GlobalUserSpecifiedConfigNamespace).Get("unsupported-cert-rotation-config", metav1.GetOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return true, nil
```
Comment (mfojtik) [inline pkg/operator/starter.go]: is this just retrying for connection errors?

File: pkg/operator/starter.go (hunk 1 above)
Comment (soltysh) [reply pkg/operator/starter.go]: Yes, we have identical code in kas-o's starter.go

File: pkg/operator/starter.go (hunk 4 above)
Comment (soltysh) [reply pkg/operator/starter.go]: That is correct.

File: pkg/operator/certrotationcontroller/certrotationcontroller.go (hunk 3 above)
Comment (soltysh) [reply pkg/operator/certrotationcontroller/certrotationcontroller.go]: Discussed on slack, I'll be looking into that dynamic reload of kcm certs in kas.

File: pkg/operator/starter.go [hunk 5]
```diff
@@ -139,12 +140,31 @@ func RunOperator(ctx *controllercmd.ControllerContext) error {
...
+			}
+			return false, err
+		}
+		if value, ok := certRotationConfig.Data["base"]; ok {
+			certRotationBase, _ = time.ParseDuration(value)
```
Comment (deads2k) [inline pkg/operator/starter.go]: make this fail, so we know

**Pull request merged**